LibraryApplication/LibraryApplication/Site.Master.cs

```diff
@@ -0,0 +1,43 @@
+﻿using LibraryApplication.Data;
+using System;
+using System.Web.Security;
+using System.Web.UI.WebControls;
+
+namespace LibraryApplication
+{
+    public partial class Site : System.Web.UI.MasterPage
+    {
+        private const string GetNameQuery = @"
+            SELECT FirstName + ' ' + LastName AS Name
+            FROM Employee
+            WHERE Username = @Username
+        ";
+        protected void Page_Load(object sender, EventArgs e)
+        {
+            if (!IsPostBack)
+            {
+                SetNavLinks();
+            }
+        }
+
+        protected void LogoutButton_Command(object sender, CommandEventArgs e)
+        {
+            FormsAuthentication.SignOut();
+            Response.Redirect("~/Default.aspx");
+        }
+
+        /// <summary>
+        /// This function programmatically sets the URLs for the HyperLinks in the navigation menu.
+        /// This would not be necessary if Web Forms wasn't being so difficult and disallowing 
+        /// the use of inline displaying expressions in HyperLink controls.
+        /// </summary>
```

Comment:
You should be able to use a binding expression to the set the `HyperLink.NavigateUrl` property. Happy to debug with you if you want to take a look at it in person.